TopAnalysis/src/FCNCTriLeptonCppWorker.cc

```diff
@@ -240,23 +240,17 @@ bool FCNCTriLeptonCppWorker::analyze() {
     if ( nGoodMuons     < 1 ) out_GoodLeptonCode -=  10;
     if ( nGoodElectrons < 1 ) out_GoodLeptonCode -= 100;
     for ( unsigned i=0; i<4; ++i ) {
-	if ( nGoodMuons     >= 1 && in_Muons_p4[0]->At(muonIdxs[0]) > LeadingLep_pt ) {
-	    out_Lepton2_p4[i] = in_Muons_p4[i]->At(muonIdxs[0]);
-	    if ( nGoodMuons     >= 2 && in_Muons_p4[0]->At(muonIdxs[1]) > RestLep_pt )
-		out_Lepton3_p4[i] = in_Muons_p4[i]->At(muonIdxs[1]);
-	    if ( nGoodElectrons >= 1 && in_Electrons_p4[0]->At(electronIdxs[0]) > RestLep_pt )
-		out_Lepton1_p4[i] = in_Electrons_p4[i]->At(electronIdxs[0]);
-	}
-    }
-    if ( nGoodMuons     >= 1 && in_Muons_p4[0]->At(muonIdxs[0]) > LeadingLep_pt ) {
-	out_Lepton2_pdgId = -13*in_Muons_charge->At(muonIdxs[0]);
-	if ( nGoodMuons     >= 2 && in_Muons_p4[0]->At(muonIdxs[1]) > RestLep_pt )
-	    out_Lepton3_pdgId = -13*in_Muons_charge->At(muonIdxs[1]);
-	if ( nGoodElectrons >= 1 && in_Electrons_p4[0]->At(electronIdxs[0]) > RestLep_pt )
-	    out_Lepton1_pdgId = -11*in_Electrons_charge->At(electronIdxs[0]);
+	if ( nGoodMuons     >= 1 ) out_Lepton2_p4[i] = in_Muons_p4[i]->At(muonIdxs[0]);
+	if ( nGoodMuons     >= 2 ) out_Lepton3_p4[i] = in_Muons_p4[i]->At(muonIdxs[1]);
+	if ( nGoodElectrons >= 1 ) out_Lepton1_p4[i] = in_Electrons_p4[i]->At(electronIdxs[0]);
     }
+    if ( nGoodMuons     >= 1 ) out_Lepton2_pdgId = -13*in_Muons_charge->At(muonIdxs[0]);
+    if ( nGoodMuons     >= 2 ) out_Lepton3_pdgId = -13*in_Muons_charge->At(muonIdxs[1]);
+    if ( nGoodElectrons >= 1 ) out_Lepton1_pdgId = -11*in_Electrons_charge->At(electronIdxs[0]);
+    
```

Comment:
In different flavour channel, Pt of leading and rest leptons are checked (line 254~265) after leptons are filled in out_Lepton_p4 list.